src/page.rs

```diff
@@ -446,48 +462,52 @@ impl<H: Header> PageWriter<H> {
 
         self.erase_if_needed(flash).await.map_err(Error::Flash)?;
 
-        let align_offs = self.chunk_pos % ALIGN;
-        if align_offs != 0 {
-            let left = ALIGN - align_offs;
-            let n = left.min(data.len());
-
-            self.align_buf[align_offs..][..n].copy_from_slice(&data[..n]);
-            data = &data[n..];
-            self.total_pos += n;
-            self.chunk_pos += n;
+        for mut data in data.chunks(MAX_CHUNK_SIZE) {
+            let align_offs = self.chunk_pos % ALIGN;
+            if align_offs != 0 {
+                let left = ALIGN - align_offs;
+                let n = left.min(data.len());
+
+                self.align_buf[align_offs..][..n].copy_from_slice(&data[..n]);
+                data = &data[n..];
+                self.total_pos += n;
+                self.chunk_pos += n;
+
+                if n == left {
+                    flash
+                        .write(
+                            self.page_id as _,
+                            self.chunk_offset + ChunkHeader::SIZE + self.chunk_pos - ALIGN,
+                            &self.align_buf,
+                        )
+                        .await
+                        .map_err(Error::Flash)?;
+                }
+            }
 
-            if n == left {
+            let n = data.len() - (data.len() % ALIGN);
+            if n != 0 {
                 flash
                     .write(
                         self.page_id as _,
-                        self.chunk_offset + ChunkHeader::SIZE + self.chunk_pos - ALIGN,
-                        &self.align_buf,
+                        self.chunk_offset + ChunkHeader::SIZE + self.chunk_pos,
+                        &data[..n],
                     )
                     .await
                     .map_err(Error::Flash)?;
+                data = &data[n..];
+                self.total_pos += n;
+                self.chunk_pos += n;
             }
-        }
 
-        let n = data.len() - (data.len() % ALIGN);
-        if n != 0 {
-            flash
-                .write(
-                    self.page_id as _,
-                    self.chunk_offset + ChunkHeader::SIZE + self.chunk_pos,
-                    &data[..n],
-                )
-                .await
-                .map_err(Error::Flash)?;
-            data = &data[n..];
+            let n = data.len();
+            assert!(n < ALIGN);
+            self.align_buf[..n].copy_from_slice(data);
             self.total_pos += n;
             self.chunk_pos += n;
-        }
 
-        let n = data.len();
-        assert!(n < ALIGN);
-        self.align_buf[..n].copy_from_slice(data);
-        self.total_pos += n;
-        self.chunk_pos += n;
+            self.commit(flash).await?;
```

Comment:
we shouldn't commit always, only when the chunk gets actually full. The `file` layer does lots of tiny writes, committing for each will result in a ton of tiny chunks.